persistence/soft-index/src/main/java/org/infinispan/persistence/sifs/SoftIndexFileStore.java

```diff
@@ -241,7 +240,8 @@ public void stop() {
          indexQueue = null;
          storeQueue = null;
       } catch (InterruptedException e) {
-         throw new PersistenceException("Cannot stop cache store", e);
+         Thread.currentThread().interrupt();
```

Comment:
I wouldn't normally expect to reinterpret the thread when throwing an exception that stating that we were interrupted. I find I would normally reset the status and return early or throw the exception, but not both.

Reply:
OK, I have thought the best practice is to set thread status to interrupted if the thrown exception is not directly `InterruptedException`.

Reply:
Yeah I was thinking it was a different exception. Seems good.